src/backends/caffe/caffeinputconns.cc

```diff
@@ -145,7 +153,42 @@ namespace dd
 	++cl;
 	++uit;
       }
-    if (_shuffle)
+		}
+		// Else if input is a file
+		else
+		{
+			std::ifstream infile(rpaths.at(0));
+			std::string line;
+			int line_num = 1;
+			int cl = 0;
+			while (std::getline(infile, line))
+			{
+				std::istringstream iss(line);
+				string filename;
+				string label;
+				CHECK(iss >> filename) << "Error reading line " << line_num;
```

Comment:
If this is the CHECK from Caffe, let's avoid it (as it's overriden across backends), and use a straight `if`.

Reply:
I removed the CHECK as it will be catched anyway.